src/main/java/com/amazon/opendistroforelasticsearch/security/auth/UserInjector.java

```diff
@@ -66,33 +68,30 @@
 
     }
 
-    boolean injectUser(RestRequest request) {
-
+    private User getUser(String injectedUserString) {
```

Comment:
Introduce `private static InjectedUser` class that extends `User`. Add `TransportAddress` as member of `InjectedUser`. Add `setTransportAddress(String ipandport)` to `InjectedUser`. Rename getUser() to `getInjectedUser()` and change return type from `User` to `InjectedUser`.

Reply:
Defined InjectedUser class for injected user usecase